rest_tools/server/handler.py

```diff
@@ -93,6 +94,10 @@ def initialize(self, debug=False, auth=None, auth_url=None, module_auth_key='',
         self.server_header = server_header
         self.route_stats = route_stats
 
+    @tracing_tools.spanned(kind="server", these=["self.request.method", "self.request.uri"])
+    async def _execute(self, *args: Any, **kwargs: Any) -> None:
+        return await super()._execute(*args, **kwargs)
```

Comment:
I'm unsure how I feel about overriding a "private" underscore method.  Is there a good reason to take this approach instead of using prepare/on_finish?

Reply:
I agree it's not great that it's private.

This allows downstream spans can inherit from the global "current span context". I suppose I could manually enter/exit a `contextmanager` instance, instead. I want to avoid using an instance variable to store the span.

Reply:
If it's just the manual enter/exit, I'd suggest doing that.  The one thing `_execute` gives you is async.  So, if you need async in the decorator then it's worth it.

Reply:
I'll try it. I don't suspect async/sync will be much of a problem since the calls to enter/exit are themselves synchronous.

Reply:
It was a bust. But, I got to refactor the `inject` argument which is a plus. I'll have to come up with a plan C.

Reply:
Here's Plan C:
1. Open an "manual span" in `prepare()` via `spanned(kind=SpanKind.SERVER, ..., behavior=SpanBehavior.MANUAL_CURRENT_SPAN)` (dev needed). This creates and sets a "current span" via `trace.set_span_in_context(span)` BUT does not end the span AND _avoids using a contextmanager altogether_.
2. Request methods (`get()`, `post()`, etc.) don't _need_ to do anything. From their POV, there's a current span from which they can create child spans.
3. End the current span in `on_finish()` via `get_current_span().end()`

If this fails, Plan D: Use "span reuse". This successively and intermittently enters/exits context managers with a shared span instance over multiple uses, via `trace.use_span()`. I'm unsure if this maintains the "current span" between explicit calls to `trace.use_span()`. So I'm hoping Plan C works.

Reply:
Is plan A the `_execute()` method? That may be better than plan D.

Reply:
Yes, and yes

Reply:
Plan C is a no-go. Plan D would require explicitly decorating the http-method handlers with a reference to a persisted span instance, which is worse than _pre-plan A_.

Plan A is the best way to do this. There's no way to give http-method handlers (`get()`, etc.) access to the "current span", other than creating a "current span" somewhere above in its call stack. I go to some more length about this in: https://github.com/WIPACrepo/wipac-telemetry-prototype/pull/11.